Source/WebCore/platform/graphics/FontPlatformData.cpp

```diff
@@ -106,6 +106,12 @@ const FontPlatformData::CreationData* FontPlatformData::creationData() const
     return m_customPlatformData ? &m_customPlatformData->creationData : nullptr;
 }
 
+float FontPlatformData::usedFontSize(float computedFontSize, const FontSizeAdjust& fontSizeAdjust, float fontFaceSizeAdjust)
+{
+    // It is not worth modifying the used size with @font-face size-adjust if we are to re-adjust it later with font-size-adjust. This is because font-size-adjust will overrule this change, since size-adjust also modifies the font's metric values and thus, keeps the aspect-value unchanged. This allows us to use the used value as computed value later on updateSizeWithFontSizeAdjust(), since they will be the same for a valid font-size-adjust.
+    return fontSizeAdjust.value ? computedFontSize : fontFaceSizeAdjust * computedFontSize;
```

Comment:
What happens when the `font-size-adjust` property and the `font-size-adjust` descriptor disagree? Who wins?

How does this interact with -webkit-text-size-adjust (on both iPhone and iPad, which use different algorithms?) I think we need some WebKit-specific tests for these interactions.

Reply:
`font-size-adjust` should win because `size-adjust` doesn't affect the aspect ratio, and font-size-adjust will act on the computed font size, which is not altered.

Reply:
> What happens when the font-size-adjust property and the font-size-adjust descriptor disagree? Who wins?

That is clarified by the level 5 spec. https://www.w3.org/TR/css-fonts-5/#font-resources
> Note: ... Consequently, applying font-size-adjust and size-adjust together means that size-adjust appears to have no effect.

It might be better to have the pointer in the comment here.